google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -622,7 +651,8 @@ public void onSuccess(long[] result) {}
   @Override
   public ApiFuture<Void> writeAsync(Iterable<Mutation> mutations) {
     Preconditions.checkNotNull(mutations);
-    checkValidTransaction();
+    // We actually don't need a transaction yet, as mutations are buffered until commit.
+    checkValidTransaction(null);
```

Comment:
nit: can we avoid passing `null` here? Do we need this call to set the `transactionStarted` or can we avoid this call?

Reply:
We need to set the `transactionStarted`, and we need to verify that the transaction is in a valid state, so we can't skip it altogether. Also; the method takes a `ParsedStatement` as an input argument so it can check the type of statement, and we don't have a SQL statement that is equivalent with writing/buffering mutations.

What we can however do is slightly rearrange the methods so we have a separate method only for verifying the valid state of the transaction and marking it as started, but skipping the creation of the underlying transaction. I've changed it to that.